service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/CloneWorkspaceFlight.java

```diff
@@ -58,68 +32,7 @@ public CloneWorkspaceFlight(FlightMap inputParameters, Object applicationContext
 
     addStep(new CreateIdsForFutureStepsStep());
 
-    // Only create a cloud context if the source workspace has a cloud context
-    Optional<GcpCloudContext> gcpCloudContextOptional =
-        flightBeanBag.getGcpCloudContextService().getGcpCloudContext(sourceWorkspaceId);
-    if (gcpCloudContextOptional.isPresent()) {
-      SpendProfile gcpSpendProfile =
-          getSpendProfile(spendProfile, gcpCloudContextOptional.get(), flightBeanBag, userRequest);
-
-      addStep(
-          new LaunchCreateCloudContextFlightStep(
-              flightBeanBag.getWorkspaceService(),
-              CloudPlatform.GCP,
-              gcpSpendProfile,
-              ControlledResourceKeys.CREATE_GCP_CLOUD_CONTEXT_FLIGHT_ID),
-          cloudRetryRule);
-      addStep(
-          new AwaitCreateCloudContextFlightStep(
-              ControlledResourceKeys.CREATE_GCP_CLOUD_CONTEXT_FLIGHT_ID),
-          longCloudRetryRule);
-    }
-
-    Optional<AzureCloudContext> azureCloudContextOptional =
-        flightBeanBag.getAzureCloudContextService().getAzureCloudContext(sourceWorkspaceId);
-    if (azureCloudContextOptional.isPresent()) {
-      SpendProfile azureSpendProfile =
-          getSpendProfile(
-              spendProfile, azureCloudContextOptional.get(), flightBeanBag, userRequest);
-      addStep(
-          new LaunchCreateCloudContextFlightStep(
-              flightBeanBag.getWorkspaceService(),
-              CloudPlatform.AZURE,
-              azureSpendProfile,
-              ControlledResourceKeys.CREATE_AZURE_CLOUD_CONTEXT_FLIGHT_ID),
-          cloudRetryRule);
-      addStep(
-          new AwaitCreateCloudContextFlightStep(
-              ControlledResourceKeys.CREATE_AZURE_CLOUD_CONTEXT_FLIGHT_ID),
-          cloudRetryRule);
-    }
-
     addStep(new LaunchCloneAllResourcesFlightStep(), cloudRetryRule);
     addStep(new AwaitCloneAllResourcesFlightStep(), longCloudRetryRule);
   }
-
-  /*
-   * There is an API mismatch in the V1 APIs. If a new spend profile is passed with
-   * the clone workspace, then we use it to create all the new cloud contexts,
-   * even though a spend profile is only valid for one cloud. Since we don't have
-   * multi-cloud workspaces right now, we will not have a mismatch of spend profile
-   * and cloud context.
-   */
-  private SpendProfile getSpendProfile(
-      SpendProfile spendProfile,
-      CloudContext cloudContext,
-      FlightBeanBag flightBeanBag,
-      AuthenticatedUserRequest userRequest) {
-    if (spendProfile != null) {
-      return spendProfile;
-    }
-
-    var spendProfileService = flightBeanBag.getSpendProfileService();
-    var features = flightBeanBag.getFeatureConfiguration();
-    return spendProfileService.authorizeLinking(
```

Comment:
This checking was actually also happening [earlier](https://github.com/DataBiosphere/terra-workspace-manager/blob/cfdaa8e366001b33b2cfa0a8c63710f1fd8e37af/service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java#L623) in the code execution path.